### PR TITLE
await: 1.0.1 -> 1.0.2

### PR DESCRIPTION
**Pull request by chewblacka**
## Description of changes
https://github.com/slavaGanzin/await/releases/tag/v1.0.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NickCao** [comment]
Shell completion file names are wrong.
```
result/share/
├── bash-completion
│   └── completions
│       └── autocomplete.bash
├── doc
│   └── await
│       └── README.md
├── fish
│   └── vendor_completions.d
│       └── autocomplete.fish
├── licenses
│   └── await
│       └── LICENSE
└── zsh
    └── site-functions
        └── _autocomplete
```

**Comment by chewblacka** [comment]
> Shell completion file names are wrong.

Apologies! Should be good now:

    result/share/
        ├── bash-completion
        │   └── completions
        │       └── await.bash
        ├── doc
        │   └── await
        │       └── README.md
        ├── fish
        │   └── vendor_completions.d
        │       └── await.fish
        ├── licenses
        │   └── await
        │       └── LICENSE
        └── zsh
            └── site-functions
                └── _await

File: pkgs/by-name/aw/await/package.nix [hunk 1]
```diff
@@ -27,6 +30,10 @@ stdenv.mkDerivation rec {
     install -Dm755 await -t $out/bin
     install -Dm444 LICENSE -t $out/share/licenses/await
     install -Dm444 README.md -t $out/share/doc/await
+    installShellCompletion --cmd await \
+      --bash <($out/bin/await --autocomplete-bash) \
+      --fish <($out/bin/await --autocomplete-fish) \
+      --zsh <($out/bin/await --autocomplete-zsh)
```
Comment (NickCao) [inline pkgs/by-name/aw/await/package.nix:36]:
```suggestion
    installShellCompletion --cmd await autocomplete.{bash,fish,zsh}
```

Using the pregenerated files is still preferred, just need to set the cmd flag.

**Review by NickCao — APPROVED** [review]

**Comment by slavaGanzin** [comment]
@chewblacka Please generate completions with --completions-zsh. 

I will remove autocomplete files from repo soon, because they are duplicating. Or should I add them to releases?

**Comment by NickCao** [comment]
> @chewblacka Please generate completions with --completions-zsh.
> 
> I will remove autocomplete files from repo soon, because they are duplicating. Or should I add them to releases?

Problem with generating completion file at build time is when cross compiling, you can't run the freshly built binary (unless with an emulator like qemu). I see that the completion files are just hardcoded in the source code, can we use some pre processor magic to embed them instead (Until we get C23)?

**Comment by slavaGanzin** [comment]
git actions allowed running freshly build app. So I just packaged them together. Don't know will it work for nix

**Comment by NickCao** [comment]
> git actions allowed running freshly build app. So I just packaged them together. Don't know will it work for nix
> 
> https://github.com/slavaGanzin/await/releases/tag/v1.0.4

That means we would have to fetch 3 additional files in addition to a git repo, three more hashes to update :(

I would say keeping the duplicated files is not a bad thing, they do simplify packing for not only nixpkgs, but every distro out there.

**Comment by slavaGanzin** [comment]
Good point. Thanks for an explanation

Will leave them there https://github.com/slavaGanzin/await/tree/main/autocompletions

**Pull request merged**